### DISCUSSION TITLE
pyemu disv example

### DISCUSSION THREAD

**Discussion by Paszka1** [pypest/pyemu#419] [General]
I wonder if there is an axample/tutorial on pyemu for mf6 disv (quadtree) grids? The example folder contains a quadtree version of the Freyberg model, but I could not find any of the notebooks being related to it. Any help on this matter would be greatly appreciated.

**Comment by Paszka1** [comment]
What I'm struggling with is adding parameters. For some reason the input file structure is not accepted as the last row has less columns than all other rows. I get a message stating this (i.e., `the number of columns changed from 20 to 4 at row 535; use `usecols` to select a subset and avoid this error`. This shouldn't be a problem with disv grids though. The idomain shape matches the number of values in the input file. I've tried different pp spacing options for `add_parameters` such as different pp_space options (float and pd.DataFrame), but no luck. Below is the code fragment I'm using and the input file is attached as well.

```
pp = pyemu.utils.pp_utils.get_zoned_ppoints_for_vertexgrid(spacing=2500.0,
                                                           zone_array=ib,
                                                           mg=gwf.modelgrid,
                                                           )

name = [f'pp1_{i:04d}' for i in range(len(pp))]
x, y = zip(*pp)
pp_df = pd.DataFrame({"name": name,
                      "x": x,
                      "y": y,
                      "zone": np.ones(len(pp), dtype=int).tolist()
                     }
                    ).set_index('name')

f = 'royval6.npf_hkx_lay_01.txt'

df_pp = pf.add_parameters(f,
                          zone_array=ib,
                          par_type="pilotpoint",
                          geostruct=grid_gs,
                          par_name_base=f.split(".")[1].replace("_", "")+"pp",
                          pargp=f.split(".")[1].replace("_", "")+"pp",
                          lower_bound=0.25, upper_bound=2.5,
                          ult_lbound=0.5, ult_ubound=5.0,
                          pp_space=pp_df,
                         )
```

[royval6.npf_hkx_lay_01.txt](https://github.com/pypest/pyemu/files/11214911/royval6.npf_hkx_lay_01.txt)

**Reply by jtwhite79** [reply]
I think you need to flatten that hk array file from a jagged 2d layout to a 1d vector layout

**Reply by Paszka1** [reply]
Thanks Jeremy. It worked. 

I suggest though to make changes to the code instead of the modeler being required to modify the standard MODFLOW input file and write a new file.
